onnxscript/function_libs/torch_aten/param_manipulation.py

```diff
@@ -0,0 +1,171 @@
+    onnx.defs.OpSchema.AttrType.GRAPHS: None,
+    onnx.defs.OpSchema.AttrType.SPARSE_TENSORS: None,
+    onnx.defs.OpSchema.AttrType.TYPE_PROTOS: None,
+}
+
+
+def _get_attribute_value(attr_proto):
+    if attr_proto.type == onnx.AttributeProto.UNDEFINED:
+        return _EmptyDefault
+    if attr_proto.type == onnx.AttributeProto.FLOAT:
+        return attr_proto.f
+    if attr_proto.type == onnx.AttributeProto.INT:
+        return attr_proto.i
+    if attr_proto.type == onnx.AttributeProto.STRING:
+        return attr_proto.s
+    if attr_proto.type == onnx.AttributeProto.FLOATS:
+        return [float(v) for v in attr_proto.f]
+    if attr_proto.type == onnx.AttributeProto.INTS:
+        return [int(v) for v in attr_proto.i]
+    raise TypeError(f"Unsupported attribute type: {attr_proto.type}")
+
+
+def extract_param_schema_from_op_schema(op_schema: onnx.defs.OpSchema) -> List[ParamSchema]:
+    param_schemas = []
+    for input_ in op_schema.inputs:
+        param_schema = ParamSchema(name=input_.name, is_input=True)
+        param_schemas.append(param_schema)
+    for attr_name, attribute in op_schema.attributes.items():
+        default_attr_proto = attribute.default_value
+        param_schema = ParamSchema(
+            name=attr_name,
+            type=_ATTRIBUTE_TYPE_TO_PYTHON_TYPE[attribute.type],
+            default=_get_attribute_value(default_attr_proto),
+            is_input=False,
+        )
+        param_schemas.append(param_schema)
+
+    return param_schemas
+
+
+def separate_input_attributes_from_arguments(
+    param_schemas: Sequence[ParamSchema],
+    args,
+    kwargs,
+) -> Tuple[List[Any], OrderedDict[str, Any]]:
+    """Separate Python args and kwargs into ONNX inputs and attributes.
+
+    Args:
+        param_schemas: The parameter schemas of an Op or a OnnxFunction.
+        args: The Python positional arguments supplied by the caller.
+        kwargs: The Python keyword arguments supplied by the caller.
+
+    Returns:
+        A tuple of two elements:
+        - A list of ONNX inputs.
+        - An ordered dictionary of ONNX attribute names and values.
+    """
+    # args, kwargs and param_schemas should be all in order
+    # user might not specify all attributes
+    if len(args) + len(kwargs) > len(param_schemas):
+        raise TypeError("Inputs are more than expected in schema")
+
+    onnx_inputs = []
+    onnx_attributes = OrderedDict()
+    for i, param in enumerate(param_schemas):
+        if i < len(args):
+            if not param.is_attribute:
+                onnx_inputs.append(args[i])
+            else:
+                onnx_attributes[param.name] = args[i]
+        elif param.name in kwargs:
+            if not param.is_attribute:
+                onnx_inputs.append(kwargs[param.name])
+            else:
+                onnx_attributes[param.name] = kwargs[param.name]
+        else:
+            # input doesn't have default
+            onnx_attributes[param.name] = param.default
```

Comment:
I haven't seen the usage of this function. Perhaps some usage contexts would not want default-values to be filled in. Anyway, I guess we can cross that bridge when we come to it.

Reply:
I added a fill_defaults option to the function